repository-structure.md

```diff
@@ -0,0 +1,72 @@
+# Common Repository Structure
+
+Hyperledger projects are required to maintain a standard set of files in each repository. This
+document describes the required and recommended files.
+
+## Required Files with Specified Content
+
+Repositories MUST have these files with the specific content in the linked files, or a file with a
+link to the specified content with minimal exposition. These files MUST be at the root of the
+repository.
+
+-   [`LICENSE`](https://www.apache.org/licenses/LICENSE-2.0.txt) - https://www.apache.org/licenses/LICENSE-2.0.txt
+-   [`CODE_OF_CONDUCT.md`](https://wiki.hyperledger.org/community/hyperledger-project-code-of-conduct) - https://wiki.hyperledger.org/community/hyperledger-project-code-of-conduct
+-   [`SECURITY.md`](https://wiki.hyperledger.org/display/SEC/Defect+Response) - https://wiki.hyperledger.org/display/SEC/Defect+Response
+
+## Required Files with Variable Content
+
+Repositories MUST have these files. Named files MUST be at the root of the repository, and may have
+format suffixes such as `.md`, `.rst`, or `.txt`.
+
+-   `README` \
+    A description of the project and contain information or links to information such as
+    -   A reference to the Apache license (required).
+    -   The current and important past releases
+    -   Documentation for developers and users
+-   `MAINTAINERS` \
+    A list of all current maintainers with contact info. [A separate document](MAINTAINERS-guidelines.md)
+    covers the specifics.
+-   `CONTRIBUTING` \
+    Directions on how to contribute code to the project, or a pointer to the Wiki page with that information.
+-   `CHANGELOG` \
```

Comment:
`Hyperledger Sawtooth` project requested that `CHANGELOG` file be optional. The project follows elaborate release process which includes creating a tag, release notes and the commit history is pretty clean. Please refer to https://github.com/hyperledger-labs/hyperledger-community-management-tools/pull/67

Reply:
I think that's too substantial a change to move with only two TSC members reviewing a PR.  When the TSC as a whole voted on it it was included as required.

My take is that in this case the CHANGELOG could stay and point to the Github repository changes list.  However this reduces the utility of the file when the repo is cloned by a non HLP project.

Reply:
Happy to include this as a subject for amendment to the earlier proposal.

Reply:
TSC on 2011-07-15 was no action.

Reply:
Outcome: File is necessary, this file can optionally redirect folks to another location where release notes are maintained.